mne/simulation/raw.py

```diff
@@ -80,9 +79,21 @@ def simulate_raw(raw, stc, trans, src, bem, cov='simple',
         and ``first_samp`` properties will be used.
         Can be ``info`` if ``duration`` is also supplied, in which case
         ``first_samp`` will be set to zero.
-    stc : instance of SourceEstimate
+
+        .. versionchanged:: 0.18
+           Support for :class:`mne.Info`.
+    stc : instance of SourceEstimate | tuple | iterable
         The source estimate to use to simulate data. Must have the same
-        sample rate as the raw data.
+        sample rate as the raw data. Can also be a tuple of
+        ``(SourceEstimate, int)`` or ``(SourceEstimate, ndarray)``
+        to allow specifying the event number (or event channel data) to
+        accompany the source estimate.
+        Can also be an iterable of these options to allow the stc to
+        vary in time, in which case all must have the same ``stc.vertices``.
+        See Notes for details.
```

Comment:
we need to discuss this. I don't like that when you pass an iterable you get the stcs stacked in time. For you need to define events and then associate to each event an stc. I would therefore allow stc to be SourceEstimate or a dict with key -> val are int -> stc or a factory of stc. makes sense?

Reply:
No, it's not as flexible. We discussed this in length in the associated issue and this is not what was decided...

Reply:
See:

https://github.com/mne-tools/mne-python/issues/5058#issuecomment-455592481

Reply:
... and finally where @makkostya assented and (I thought!) so did you:

https://github.com/mne-tools/mne-python/issues/5058#issuecomment-456813821

Reply:
won't this be replaced with a `SourceGenerator` class anyway?

Reply:
My idea is to have the SourceGenerator class be iterable, or have a method that returns an iterable of STCs. So that would end up being the input to this function.